### PR TITLE
Add clubs to national team emails.

### PR DESCRIPTION
**Pull request by itsluketwist**

### CODE REVIEW

File: src/routes/auth.js [hunk 1]
```diff
@@ -146,9 +146,11 @@ export default function authRoute() {
   }));
 
   router.put('/national', authenticateJWT, checkAuthenticated, asyncHandler(async (req, res) => {
-    const { first_name, last_name } = await prisma.users.findUnique({ where: { uuid: req.user.uuid } });
+    const { first_name, last_name, club_uuid } = await prisma.users.findUnique({ where: { uuid: req.user.uuid } });
     await update(req.user.uuid, req.body);
 
+    const { name } = await getClub(club_uuid);
```
Comment (declanramsay) [inline src/routes/auth.js:152]: I would rename this as I think it's too easy to get confused down the line especially at the email layer, where this just comes through without context

```suggestion
    const { name: club_name } = await getClub(club_uuid);
```

File: src/routes/auth.js [hunk 2]
```diff
@@ -161,17 +163,20 @@ export default function authRoute() {
...
   }));
 
   router.put('/scouting', authenticateJWT, checkAuthenticated, asyncHandler(async (req, res) => {
     const {
-      first_name, last_name, first_team, second_team, third_team, position, playstyle, years, experience,
+      first_name, last_name, first_team, second_team, third_team, position, playstyle, years, experience, club_uuid,
     } = await prisma.users.findUnique({ where: { uuid: req.user.uuid } });
 
-    // Email to head scout, with spplication and national team profile information.
+    const { name } = await getClub(club_uuid);
```
Comment (declanramsay) [inline src/routes/auth.js:177]: See comment above about improving the context for the email, you'll also need to change the value in the payload below it

```suggestion
    const { name: club_name } = await getClub(club_uuid);
```

**Review by declanramsay — COMMENTED** [review]
Having had a look at the FE we will need to make a change to the `national-team` profile page as we currently don't check if they have registered for a club yet, which would cause this to error if they don't have a club. It's a similar check as we do on the club page where it checks for membership and redirects them if they don't.

**Comment by itsluketwist** [comment]
> Having had a look at the FE we will need to make a change to the `national-team` profile page as we currently don't check if they have registered for a club yet, which would cause this to error if they don't have a club. It's a similar check as we do on the club page where it checks for membership and redirects them if they don't.
> 
> If club is not critical though we can just add guards here to skip the fetch of the club data.

I don't see why a club would be a requirement, players can be eligible for a National Team without a club in QUK (technically), so it should work for them IMO (although this will likely be very rare). I think I can just use ?? to handle if club_name is nullish, but not sure on the actual best way to do this in JSX - so please do tell me off/fix it. 😄 

Also I've added pronouns to the scouting application form in both PRs, it's not required and won't be stored in the db, but will help with the scouting team referring to players.

**Comment by declanramsay** [comment]
The code will blow because of this line:

```js
const { name: club_name } = await getClub(club_uuid);
```

the destructuring of the response to getClub will give you a `cannot read property "name" of undefined` error if you put in a null uuid as the response will be null. Hence why you either need to add some guards here to not do that call if `club_uuid` is null.

**Comment by itsluketwist** [comment]
> The code will blow because of this line:
> 
> ```js
> const { name: club_name } = await getClub(club_uuid);
> ```
> 
> the destructuring of the response to getClub will give you a `cannot read property "name" of undefined` error if you put in a null uuid as the response will be null. Hence why you either need to add some guards here to not do that call if `club_uuid` is null.

I actually had some uncommitted code doing this, my bad. 🤦‍♂️

Anyway I was hoping to do this:
```
    if (club_uuid) {
      const { name: club_name } = await getClub(club_uuid);
    }
```

But I get the following error upon commit:
```
  153:21  error  'club_name' is assigned a value but never used  no-unused-vars
  168:18  error  'club_name' is not defined                      no-undef
  180:21  error  'club_name' is assigned a value but never used  no-unused-vars
  198:18  error  'club_name' is not defined                      no-undef

✖ 4 problems (4 errors, 0 warnings)

husky > pre-commit hook failed (add --no-verify to bypass)
```

**Comment by declanramsay** [comment]
You're close but this is I think the right solution:

```js
    let club_name = null;
    if (club_uuid) {
      const club = await getClub(club_uuid);
      club_name = club.name;
    }
```
(`var` is now relatively obsolete in modern JS, `let` and `const` all the way)

**Comment by itsluketwist** [comment]
@declanramsay okay all done I think 🙌

Happy for you to just release both sets of changes when you sign them off. 😄

**Review by declanramsay — APPROVED** [review]

**Pull request merged**